lib/Echidna/Output/JSON.hs

```diff
@@ -0,0 +1,121 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module Echidna.Output.JSON where
+
+import Echidna.ABI (ppAbiValue)
+import qualified Echidna.Campaign as C
+import Echidna.Solidity (SolTest)
+import Echidna.Transaction (Tx(..), TxCall(..))
+import Data.Aeson hiding (Error)
+import qualified Data.ByteString.Base16 as BS16
+import Data.ByteString.Lazy (ByteString)
+import Data.Text
+import Data.Text.Encoding (decodeUtf8)
+
+data Top = Top
+  { _success :: Bool
+  , _error :: Maybe String
+  , _tests :: [Test]
+  , seed :: Int
+  }
+
+instance ToJSON Top where
+  toJSON Top{..} = object
+    [ "success" .= _success
+    , "error" .= _error
+    , "tests" .= _tests
+    , "seed" .= seed
+    ]
+
+data Test = Test
+  { contract :: Text
+  , name :: Text
+  , status :: TestStatus
+  , _error :: Maybe String
+  , testType :: TestType
+  , transactions :: Maybe [Transaction]
+  }
+
+instance ToJSON Test where
+  toJSON Test{..} = object
+    [ "contract" .= contract
+    , "name" .= name
+    , "status" .= status
+    , "error" .= _error
+    , "type" .= testType
+    , "transactions" .= transactions
+    ]
+
+data TestType = Property | Assertion
+
+instance ToJSON TestType where
+  toJSON Property = "property"
+  toJSON Assertion = "assertion"
+
+data TestStatus = Fuzzing | Shrinking | Solved | Passed | Error
+
+instance ToJSON TestStatus where
+  toJSON Fuzzing = "fuzzing"
+  toJSON Shrinking = "shrinking"
+  toJSON Solved = "solved"
+  toJSON Passed = "passed"
+  toJSON Error = "error"
+
+
+data Transaction = Transaction
```

Comment:
Can you restore the coverage information as we had it in our json file?

Reply:
done